### PR TITLE
Tobin event querypool pipeline invalid cb

### PR DESCRIPTION
**Pull request by tobine**
Added invalid cmd buffer cases for destroyed event, querypool, or pipeline that are bound to cmd buffer at time of destroy.

### CODE REVIEW

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -6264,6 +6264,36 @@ TEST_F(VkLayerTest, InvalidCmdBufferImageDestroyed) {
...
+    ASSERT_NO_FATAL_FAILURE(InitState());
+
+    VkEvent event;
+    VkEventCreateInfo evci = {};
+    evci.sType = VK_STRUCTURE_TYPE_EVENT_CREATE_INFO;
+    VkResult result = vkCreateEvent(m_device->device(), &evci, NULL, &event);
+    ASSERT_VK_SUCCESS(result);
+
+    m_commandBuffer->BeginCommandBuffer();
+    vkCmdSetEvent(m_commandBuffer->GetBufferHandle(), event,
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: You use `GetBufferHandle()` here, and `handle()` later?

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -3729,6 +3729,13 @@ static bool addCmd(layer_data *my_data, GLOBAL_CB_NODE *pCB, const CMD_TYPE cmd,
     }
     return skip_call;
 }
+// Tie the VK_OBJECT to the cmd buffer which includes:
+//  Add object_binding to cmd buffer
+//  Add cb_binding to object
+static void addCommandBufferBinding(std::unordered_set<GLOBAL_CB_NODE *> *cb_bindings, VK_OBJECT obj, GLOBAL_CB_NODE *cb_node) {
+    cb_bindings->insert(cb_node);
+    cb_node->object_bindings.insert(obj);
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: This is a little different to the code it's replacing -- the binding is bidirectional?

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -5412,8 +5420,16 @@ DestroyShaderModule(VkDevice device, VkShaderModule shaderModule, const VkAlloca
...
-    get_my_data_ptr(get_dispatch_key(device), layer_data_map)->device_dispatch_table->DestroyPipeline(device, pipeline, pAllocator);
-    // TODO : Clean up any internal data structures using this obj.
+    layer_data *dev_data = get_my_data_ptr(get_dispatch_key(device), layer_data_map);
+    dev_data->device_dispatch_table->DestroyPipeline(device, pipeline, pAllocator);
+
+    auto pipe_node = getPipeline(dev_data, pipeline);
+    if (pipe_node) {
+        // Any bound cmd buffers are now invalid
+        invalidateCommandBuffers(pipe_node->cb_bindings,
+                                 {reinterpret_cast<uint64_t &>(pipeline), VK_DEBUG_REPORT_OBJECT_TYPE_PIPELINE_EXT});
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: This cast noise is a bit awful...

**Comment by chrisforbes** [comment]
:+1:

**Comment by tobine** [comment]
cherry-picked and pushed to master

**Pull request closed without merging**